src/simulators/statevector/qubitvector.hpp

```diff
@@ -400,6 +404,10 @@ class QubitVector {
   size_t data_size_;
   std::complex<data_t>* data_;
   std::complex<data_t>* checkpoint_;
+  // If data_t is double, SIMD function is enable
+#ifdef SIMD
+  bool type_double = ((typeid(data_t) == typeid(double)) ? true : false);
```

Comment:
No. this code only works with double precession. So if the type is single precession, it executes Non-SIMD apply_matrix function.

Reply:
I think you could perform similar operations in single-precision with anything as new as Haswell.  https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=4919,2549,2553&text=single&techs=AVX,FMA .  Is the restriction to Macs because we think that these chips will be "new enough" Intel chips.

Reply:
Ok. I try to add simd functions for single-precision.